src/optimalcontrol/include/iDynTree/OptimalControlProblem.h

```diff
@@ -42,13 +44,12 @@ namespace iDynTree {
 
             OptimalControlProblem(const OptimalControlProblem& other) = delete;
 
-            // startingTime >= 0
             bool setTimeHorizon(double startingTime, double finalTime);
             double initialTime() const;
             double finalTime() const;
 
             bool setDynamicalSystemConstraint(std::shared_ptr<DynamicalSystem> dynamicalSystem);
-            const std::weak_ptr<const iDynTree::optimalcontrol::DynamicalSystem> dynamicalSystem() const;
+            const std::weak_ptr<DynamicalSystem> dynamicalSystem() const;
```

Comment:
Was removing `const` done on purpose? Just checking, sometimes you want to remove only the namespace and you end up removing other things :sweat_smile:

Reply:
Yes. Thanks anyway :wink:. I need to call some ``DynamicalSystem`` methods like ``dynamics`` from that pointer (which is not ``const``), or to copy it in another shared pointer.